posydon/interpolation/profile_interpolation.py

```diff
@@ -0,0 +1,572 @@
+        self.c_ind = self.model_IF.interpolators[0].out_keys.index("S1_center_h1")
+        self.s_ind = self.model_IF.interpolators[0].out_keys.index("S1_surface_h1")
+        self.interp = self.model_IF.interpolators[0]
+        
+        # star 1 states
+        self.class_names = ['None','WD','NS','BH',
+                             'stripped_He_non_burning',
+                             'stripped_He_Core_He_burning',
+                             'stripped_He_Core_C_burning',
+                             'stripped_He_Central_He_depleted',
+                             'stripped_He_Central_C_depletion',
+                             'H-rich_non_burning',
+                             'H-rich_Central_He_depleted',
+                             'H-rich_Central_C_depletion',
+                             'H-rich_Shell_H_burning',
+                             'H-rich_Core_H_burning',
+                             'H-rich_Core_He_burning',
+                             'H-rich_Core_C_burning']
+        
+        # sort training data into classes by s1 state
+        self.sort_ind = {class_name:[] for class_name in self.class_names}
+        for i in range(len(self.s1)):
+            state = self.s1[i]
+            self.sort_ind[state].append(i)
+        
+        # sort testing data into classes by s1 state
+        self.valid_sort_ind = {class_name:[] for class_name in self.class_names}
+        for i in range(len(self.valid_s1)):
+            state = self.valid_s1[i]
+            self.valid_sort_ind[state].append(i)
+        
+        # create and train models for profile boundaries
+        self.bounds_models = self.learn_bounds()
+        
+    def learn_bounds(self):
+        """Creates and trains NNs to predict boundary points for each star 1 state.
+        Returns:
+            b_models (array-like) : dictionary containing boundary models.
+        """
+        b_models = {}
+        for state in ['WD','NS','BH',
+                      'H-rich_Central_He_depleted',
+                      'H-rich_Central_C_depletion',
+                      'H-rich_Shell_H_burning',
+                      'H-rich_Core_H_burning',
+                      'H-rich_Core_He_burning',
+                      'H-rich_Core_C_burning']:
+            
+            # identify input/output training data for class
+            indices = self.sort_ind[state]
+            inputs = self.initial[indices]
+            prof = self.profiles[indices]
+
+            # identify input/output testing data for class
+            valid_indices = self.valid_sort_ind[state]
+            valid_inputs = self.valid_initial[valid_indices]
+            valid_prof = self.valid_profiles[valid_indices]
+            
+            # calculate boundary points for training and testing data 
+            if "burning" in state:  # these classes' profile shapes have 2 boundary points
+                outs = 2
+                bounds = []
+                nonflat = []  # TODO: explain this
```

Comment:
what is nonflat?

Reply:
just added this comment: # ensures that training data only has the correct "non-flat" shape
# to avoid issues with calculating the boundary points